src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/StartupObjectValueProvider.cs

```diff
@@ -17,27 +17,36 @@ internal class StartupObjectValueProvider : InterceptingPropertyValueProviderBas
     private const string DisabledValue = "WindowsFormsWithCustomSubMain";
 
     internal const string UseWinFormsProperty = "UseWindowsForms";
+    internal const string InterceptedStartupObjectProperty = "StartupObjectVB";
     internal const string StartupObjectProperty = "StartupObject";
     internal const string RootNamespaceProperty = "RootNamespace";
 
+    private readonly UnconfiguredProject _project;
+
     [ImportingConstructor]
-    public StartupObjectValueProvider(IMyAppFileAccessor myAppXmlFileAccessor)
+    public StartupObjectValueProvider(IMyAppFileAccessor myAppXmlFileAccessor, UnconfiguredProject project)
     {
         _myAppXmlFileAccessor = myAppXmlFileAccessor;
+        _project = project;
     }
 
     public override async Task<string?> OnSetPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties, IReadOnlyDictionary<string, string>? dimensionalConditions = null)
     {
-        string isWindowsFormsProject = await defaultProperties.GetEvaluatedPropertyValueAsync(UseWinFormsProperty);
+        IProjectCapabilitiesScope capabilities = _project.Capabilities;
+        bool isWindowsForms = capabilities.Contains(ProjectCapability.WindowsForms);
         string rootNameSpace = await defaultProperties.GetEvaluatedPropertyValueAsync(RootNamespaceProperty);
 
-        if (bool.TryParse(isWindowsFormsProject, out bool b) && b)
+        if (isWindowsForms)
         {
             string applicationFrameworkValue = await defaultProperties.GetEvaluatedPropertyValueAsync(ApplicationFrameworkProperty);
 
             if (Equals(applicationFrameworkValue, EnabledValue))
             {
                 // Set the startup object in the myapp file.
+                if(unevaluatedPropertyValue.StartsWith(rootNameSpace + ".", StringComparison.OrdinalIgnoreCase))
+                {
+                    unevaluatedPropertyValue = unevaluatedPropertyValue.Substring((rootNameSpace + ".").Length);
```

Comment:
This string allocation could be avoided. You could either re-use the value computed in the `if` condition, or just `rootNameSpace.Length + 1`.